### PR TITLE
chore: rework dependency updater

### PR DESCRIPTION
**Pull request by MelvinKl**

### CODE REVIEW

File: .github/workflows/dependency-checker.yaml [hunk 1]
```diff
@@ -24,30 +24,30 @@ jobs:
...
-            done
+            if [ -n "$(git diff --name-only)" ]; then
+              for file in $(git diff --name-only | grep poetry.lock); do
+                  # Extract the service for which the dependencies have been updated
+                  dirpath=$(dirname $file)                
+                  git add "$file"
+                  git commit -m "chore: dependency update for ${dirpath}"
+              done
+              
+              # Check if a PR already exists for the package
```
Comment (joaopalet) [inline .github/workflows/dependency-checker.yaml:42]: In this case is if any other dependency update PR, right?

Comment (joaopalet) [reply .github/workflows/dependency-checker.yaml:42]:
```suggestion
              # Check if a PR already exists for dependency updates
```

File: .github/workflows/dependency-checker.yaml [hunk 2]
```diff
@@ -24,30 +24,30 @@ jobs:
             git config --global user.email "noreply@stackit.de"
         
             pip install poetry
+
+            pr_name=$(echo "Dependency Update")
```
Comment (joaopalet) [inline .github/workflows/dependency-checker.yaml:28]:
```suggestion
            pr_name=$(echo "Dependency Updates")
```

File: .github/workflows/dependency-checker.yaml (hunk 1 above)
Comment (MelvinKl) [reply .github/workflows/dependency-checker.yaml]: Yes. If there already is a PR it will be closed and replaced by the new one

**Pull request merged**